pyntc/devices/iosxewlc_device.py

```diff
@@ -0,0 +1,90 @@
+"""Module for using a Cisco IOSXE WLC device over SSH."""
+import time
+
+from .ios_device import IOSDevice
+from pyntc.errors import RebootTimeoutError, OSInstallError, InstallModeRequired
+
+INSTALL_MODE_FILE_NAME = "packages.conf"
+
+
+class IOSXEWLCDevice(IOSDevice):
+    """Cisco IOSXE WLC Device Implementation."""
+
+    def _wait_for_device_start_reboot(self):
+        while True:
+            try:
+                self.open()
+                self.show("show version")
+                return
+            except:  # noqa E722 # nosec
+                break
+
+    def _wait_for_device_reboot(self, timeout=5400):
+        start = time.time()
+        while time.time() - start < timeout:
+            try:
+                self.open()
+                self.show("show version")
+                return
+            except:  # noqa E722 # nosec
+                pass
+
+        raise RebootTimeoutError(hostname=self.hostname, wait_time=timeout)
+
+    def install_os(self, image_name, install_mode=False, install_mode_delay_factor=20, **vendor_specifics):
+        """Installs the prescribed Network OS, which must be present before issuing this command.
+
+        Args:
+            image_name (str): Name of the IOS image to boot into
+            install_mode (bool, optional): Uses newer install method on devices. Defaults to False.
+
+        Raises:
+            OSInstallError: Unable to install OS Error type
+
+        Returns:
+            bool: False if no install is needed, true if the install completes successfully
+        """
+        timeout = vendor_specifics.get("timeout", 5400)
+        if not self._image_booted(image_name):
+            if install_mode:
+                # Change boot statement to be boot system <flash>:packages.conf
+                self.set_boot_options(INSTALL_MODE_FILE_NAME, **vendor_specifics)
+
+                # Get the current fast_cli to set it back later to whatever it is
+                current_fast_cli = self.fast_cli
+
+                # Set fast_cli to False to handle install mode, 10+ minute installation
+                self.fast_cli = False
+
+                # Run install command (which reboots the device)
+                command = f"install add file {self._get_file_system()}{image_name} activate commit prompt-level none"
+                # Set a higher delay factor and send it in
+                try:
+                    self.show(command, delay_factor=install_mode_delay_factor)
+                except IOError:
```

Comment:
Why is IOError raised, and why is it being ignored?

Reply:
As discussed a comment has been added for more info on this try/except block.